ovos_utils/xdg_utils.py

```diff
@@ -0,0 +1,174 @@
+    absolute path, the default value is returned.
+
+    Parameters
+    ----------
+    variable : str
+        Name of the environment variable.
+    default : Path
+        Default value.
+
+    Returns
+    -------
+    Path
+        Value from environment or default.
+
+    """
+    # TODO(srstevenson): Use assignment expression in Python 3.8.
+    value = os.environ.get(variable)
+    if value and os.path.isabs(value):
+        return Path(value)
+    return default
+
+
+def _paths_from_env(variable: str, default: List[Path]) -> List[Path]:
+    """Read an environment variable as a list of paths.
+
+    The environment variable with the specified name is read, and its
+    value split on colons and returned as a list of paths. If the
+    environment variable is not set, or set to the empty string, the
+    default value is returned. Relative paths are ignored, as per the
+    specification.
+
+    Parameters
+    ----------
+    variable : str
+        Name of the environment variable.
+    default : List[Path]
+        Default value.
+
+    Returns
+    -------
+    List[Path]
+        Value from environment or default.
+
+    """
+    # TODO(srstevenson): Use assignment expression in Python 3.8.
+    value = os.environ.get(variable)
+    if value:
+        paths = [
+            Path(path) for path in value.split(":") if os.path.isabs(path)
+        ]
+        if paths:
+            return paths
+    return default
+
+
+def xdg_cache_home() -> Path:
+    """Return a Path corresponding to XDG_CACHE_HOME."""
+    return _path_from_env("XDG_CACHE_HOME", Path.home() / ".cache")
+
+
+def xdg_config_dirs() -> List[Path]:
+    """Return a list of Paths corresponding to XDG_CONFIG_DIRS."""
+    return _paths_from_env("XDG_CONFIG_DIRS", [Path("/etc/xdg")])
+
+
+def xdg_config_home() -> Path:
+    """Return a Path corresponding to XDG_CONFIG_HOME."""
+    return _path_from_env("XDG_CONFIG_HOME", Path.home() / ".config")
+
+
+def xdg_data_dirs() -> List[Path]:
+    """Return a list of Paths corresponding to XDG_DATA_DIRS."""
+    return _paths_from_env(
+        "XDG_DATA_DIRS",
+        [Path(path) for path in "/usr/local/share/:/usr/share/".split(":")],
+    )
+
+
+def xdg_data_home() -> Path:
+    """Return a Path corresponding to XDG_DATA_HOME."""
+    return _path_from_env("XDG_DATA_HOME", Path.home() / ".local" / "share")
+
+
+def xdg_runtime_dir() -> Optional[Path]:
+    """Return a Path corresponding to XDG_RUNTIME_DIR.
+
+    If the XDG_RUNTIME_DIR environment variable is not set, None will be
+    returned as per the specification.
+
+    """
+    value = os.getenv("XDG_RUNTIME_DIR")
+    if value and os.path.isabs(value):
+        return Path(value)
+    return None
+
+
+def xdg_state_home() -> Path:
+    """Return a Path corresponding to XDG_STATE_HOME."""
+    return _path_from_env("XDG_STATE_HOME", Path.home() / ".local" / "state")
+
+
+# The following variables are deprecated, but remain for backward compatibility.
```

Comment:
Should we exclude these deprecated variables so we don't have to support them at all?

Reply:
see comment above